bootstrap/023_simple_materializations.sql

```diff
@@ -0,0 +1,95 @@
+        let oldest_running timestamp := 0::timestamp;
+
+        if (input is not null) then
+            oldest_running := coalesce(input:oldest_running::timestamp, 0::timestamp);
+        end if;
+
+        let table_ident varchar := (select 'INTERNAL_REPORTING_MV.' || :table_name);
+        if (oldest_running = 0::timestamp) then
+          -- we should ensure that there are no records in the table if this is the first run. This allows a separate process to insert a "reset" message in the log which will cause us to start over again.
+          truncate table identifier(:table_ident);
+        end if;
+
+        -- if there are incomplete queries, find the min timestamp of the incomplete queries. If there are no incomplete, find the newest timestamp for a filter condition next time.
+        let where_clause_complete varchar := (select :index_col || ' >= to_timestamp_ltz(\'' || :oldest_running || '\')');
+        let new_closed number;
+        call internal.generate_insert_statement('INTERNAL_REPORTING_MV', :table_name, 'ACCOUNT_USAGE', :table_name, :where_clause_complete) into :new_closed;
+        let new_running timestamp := (select max(identifier(:index_col)) from identifier(:table_ident));
+        insert into INTERNAL.TASK_SIMPLE_DATA_EVENTS SELECT :dt, true, :table_name, :input, OBJECT_CONSTRUCT('oldest_running', :new_running, 'attempted_migrate', :migrate, 'new_records', coalesce(:new_closed, 0))::VARIANT;
+        COMMIT;
+
+    EXCEPTION
+      WHEN OTHER THEN
+        SYSTEM$LOG_ERROR(OBJECT_CONSTRUCT('error', 'Exception occurred while refreshing ' || :table_name || ' events.', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate));
+        ROLLBACK;
+        insert into INTERNAL.TASK_SIMPLE_DATA_EVENTS SELECT :dt, false, :table_name, :input, OBJECT_CONSTRUCT('Error type', 'Other error', 'SQLCODE', :sqlcode, 'SQLERRM', :sqlerrm, 'SQLSTATE', :sqlstate)::variant;
+        RAISE;
+
+    END;
+end;
+
+create or replace procedure internal.migrate_simple_data_view(table_name varchar) returns variant language sql as
+begin
+    let tbl varchar := 'INTERNAL_REPORTING_MV.' || :table_name;
+    let vw varchar := 'REPORTING.' || :table_name;
+    create or replace view identifier(:vw) as select * from identifier(:tbl);
+end;
+
+create table if not exists internal_reporting_mv.serverless_task_history as select * from account_usage.serverless_task_history where 1=0;
+create table if not exists internal_reporting_mv.task_history as select * from account_usage.task_history where 1=0;
+create table if not exists internal_reporting_mv.sessions as select * from account_usage.sessions where 1=0;
+create or replace view reporting.serverless_task_history as select * from account_usage.serverless_task_history where 1=0;
+create or replace view reporting.task_history as select * from account_usage.task_history where 1=0;
+create or replace view reporting.sessions as select * from account_usage.sessions where 1=0;
```

Comment:
You don't want the `where 1=0` on the view def'n, do you?

Reply:
good spot, fixed